### PR TITLE
test: [POM] Migrate transaction with snap account e2e tests to page object modal

### LINKED ISSUE #27765: [POM] Migrate transaction with snap account e2e tests to page object modal
### What is this about?

Migrate transaction with snap account e2e tests to page object modal

### Scenario

_No response_

### Design

_No response_

### Technical Details

_No response_

### Threat Modeling Framework

_No response_

### Acceptance Criteria

_No response_

### Stakeholder review needed before the work gets merged

- [ ] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by chloeYue**
## **Description**
This PR migrates the `snap-account-transfers.spec.ts` e2e tests to the Page Object Model (POM) pattern, improving test stability and maintainability.

Changes include:
- Migrate test snap-account-transfers.spec.ts to POM
- Avoid several delays in the original function implementation
- Reduce flakiness

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/27155?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/27765

## **Manual testing steps**
Check code readability, make sure tests pass.

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: test/e2e/accounts/snap-account-contract-interaction.spec.ts [hunk 1]
```diff
@@ -14,8 +14,10 @@ import {
   ACCOUNT_2,
 } from '../helpers';
 import FixtureBuilder from '../fixture-builder';
+import { installSnapSimpleKeyring } from '../page-objects/flows/snap-simple-keyring.flow';
+import { loginWithBalanceValidation } from '../page-objects/flows/login.flow';
 import { SMART_CONTRACTS } from '../seeder/smart-contracts';
-import { installSnapSimpleKeyring, importKeyAndSwitch } from './common';
+import { importKeyAndSwitch } from './common';
```
Comment (chloeYue) [inline test/e2e/accounts/snap-account-contract-interaction.spec.ts:20]: Intended change to use `installSnapSimpleKeyring` flow created following page object model.

File: test/e2e/page-objects/pages/homepage.ts [hunk 2]
```diff
@@ -6,36 +6,35 @@ import HeaderNavbar from './header-navbar';
...
+  };
 
-  private transactionAmountsInActivity: string;
+  private readonly sendButton = '[data-testid="eth-overview-send"]';
 
-  public headerNavbar: HeaderNavbar;
+  private readonly tokensTab = '[data-testid="account-overview__asset-tab"]';
+
+  private readonly transactionAmountsInActivity =
+    '[data-testid="transaction-list-item-primary-currency"]';
```
Comment (chloeYue) [inline test/e2e/page-objects/pages/homepage.ts:34]: Intended change to define and initialize variables in one line to speed up development.

**Review by DDDDDanica — APPROVED** [review]

**Comment by DDDDDanica** [comment]
LGTM !

**Comment by plasmacorral** [comment]
image

**Review by plasmacorral — APPROVED** [review]

**Pull request merged**